hw4.Rmd

```diff
@@ -30,43 +33,84 @@ Before doing anything else, just play with the data to get a sense of it.
 (@) The main outcome of interest is *decision* which records whether the judge granted an appeal or not.  In what proprotion of cases did the judge grant an appeal.
 
 ```{r}
-# [your code here]
+head(data)
+
+yes <- data %>%
+  filter(decision == "yes") %>%
+  summarise(yes <- n())
+  
+prop_yes <- yes/length(data$decision)
+prop_yes
 ```
 
 (@) There are 12 different judges in the data. A key question is whether different judges have different rates of granting an appeal.  Make a plot showing the number of cases per judge.
 
 ```{r}
-# [your code here]
+cases <- data %>%
+  group_by(judge) %>%
+  summarise(cases = n())
+p <- ggplot(data = cases, aes( x = judge, y = cases), color = "blue") + geom_bar(stat = "identity") +
+  labs(x = "Judge", y = "Number of Cases") +
+  ggtitle("Number of Cases Per Judge")
+p
+
 ```
 
 (@) Now plot the rate at which the judges grant appeals. 
 
 ```{r}
-# [your code here]
+rate_yes <- data %>%
+  group_by(judge, decision) %>%
+  filter(decision == "yes") %>%
+  summarise(yes = n()) 
+
+cases$yes <- rate_yes$yes
+
+cases <- cases %>%
+  mutate(rate_yes = (yes/cases))
+
+p1 <- ggplot(data = cases, aes(x = judge, y = rate_yes)) + 
+  geom_bar(stat = "identity") +
+  labs(x = "Judge", y = "Rate of Granting Appeals") +
+  ggtitle("Rate of Granting Appeals per Judge")
+p1
+
 ```
 
-(@) Now let's try this with logistic regression because we are going to move to more complicated models.  Fit a logistic regression predicting whetheran appeal will be granted using judge as a categorical predictor. Use Iacobucci as the reference judge to facilitate the comparisons that follow. For more on how to control the reference level of a factor, check out this post on StackOverflow: http://stackoverflow.com/questions/3872070/how-to-force-r-to-use-a-specified-factor-level-as-reference-in-a-regression
+(@) Now let's try this with logistic regression because we are going to move to more complicated models.  Fit a logistic regression predicting whether an appeal will be granted using judge as a categorical predictor. Use Iacobucci as the reference judge to facilitate the comparisons that follow. For more on how to control the reference level of a factor, check out this post on StackOverflow: http://stackoverflow.com/questions/3872070/how-to-force-r-to-use-a-specified-factor-level-as-reference-in-a-regression
 
 ```{r}
-# [your code here]
+data$judge <- relevel(as.factor(data$judge), ref="Iacobucci")
+fit <- glm(decision ~ judge, family = "binomial", data = data)
+tidy(fit)
 ```
 
 (@) Using the results of the model calculate the predicted probability that Judge Marceau will grant an appeal.
 
 ```{r}
-# [your code here]
+b <- coef(fit)
+b.marceau <- b["judgeMarceau"]
+exp(b.marceau)
+plogis(b[1] + b.marceau)
```

Comment:
Good job calling judge by name, but also try calling the intercept by name instead of position. See solutions.